wire/netaddress.go

```diff
@@ -90,12 +83,13 @@ func NewNetAddress(addr *net.TCPAddr, services ServiceFlag) *NetAddress {
 func readNetAddress(r io.Reader, pver uint32, na *NetAddress, ts bool) error {
 	var services ServiceFlag
 	var ip [16]byte
+	timestamp := na.Timestamp
```

Comment:
I've removed the allocation of the timestamp and the service flag. The benchmark is now better than before.
I thought I could also avoid the IP allocation in the case where the capacity is already 16, but adding that code makes things worse so I left that for now.

$ go test -run=NONE -bench=BenchmarkDecodeAddr -benchmem
BenchmarkDecodeAddr-4   	    2000	    517959 ns/op	   89729 B/op	    1002 allocs/op
PASS
ok  	github.com/btcsuite/btcd/wire	1.105s